airflow/providers/google/cloud/hooks/vertex_ai/pipeline_job.py

```diff
@@ -0,0 +1,414 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""This module contains a Google Cloud Vertex AI hook.
+
+.. spelling:word-list::
+
+    aiplatform
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Sequence
+
+from google.api_core.client_options import ClientOptions
+from google.api_core.gapic_v1.method import DEFAULT, _MethodDefault
+from google.cloud.aiplatform import PipelineJob
+from google.cloud.aiplatform_v1 import PipelineServiceClient
+
+from airflow.exceptions import AirflowException
+from airflow.providers.google.common.consts import CLIENT_INFO
+from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
+
+if TYPE_CHECKING:
+    from google.api_core.operation import Operation
+    from google.api_core.retry import Retry
+    from google.cloud.aiplatform.metadata import experiment_resources
+    from google.cloud.aiplatform_v1.services.pipeline_service.pagers import ListPipelineJobsPager
+
+
+class PipelineJobHook(GoogleBaseHook):
+    """Hook for Google Cloud Vertex AI Pipeline Job APIs."""
+
+    def __init__(
+        self,
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: str | Sequence[str] | None = None,
+        **kwargs,
+    ) -> None:
+        if kwargs.get("delegate_to") is not None:
+            raise RuntimeError(
+                "The `delegate_to` parameter has been deprecated before and finally removed in this version"
+                " of Google Provider. You MUST convert it to `impersonation_chain`"
+            )
```

Comment:
this is new code so why do we need handling parameters that were never defined in this class?

Reply:
I have noticed that we still have this parameter in `GoogleBaseHook` and we do not have any warning there https://github.com/apache/airflow/blob/main/airflow/providers/google/common/hooks/base_google.py#L240 . Because of that I have decided to add this message also to this code. But now I think it was a mistake. I will delete this code.